test/torchaudio_unittest/functional/functional_impl.py

```diff
@@ -227,6 +227,24 @@ def test_mask_along_axis_iid(self, mask_param, mask_value, axis):
         assert mask_specgrams.size() == specgrams.size()
         assert (num_masked_columns < mask_param).sum() == num_masked_columns.numel()
 
+    def test_resample_no_warning(self):
+        sample_rate = 44100
+        waveform = get_whitenoise(sample_rate=sample_rate)
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            F.resample(waveform, sample_rate, sample_rate / 2)
```

Comment:
I think it's better to explicitly convert to float type here.

```suggestion
            F.resample(waveform, float(sample_rate), sample_rate / 2.)
```